Hangsman/hangman.py

```diff
@@ -0,0 +1,71 @@
+class Hangman:
+
+    def __init__(self, word="Word"):
+        self.word = word.upper()
+        self._guess_count = 6
+        self. guessed_letters = []
+        self.display = ["_" for alphabet in self.word]
+
+
+    def get_guess_count(self):
+        return self._guess_count
+
+
+    def show(self):
+        shown_word = " ".join(self.display)
+        print(f"Word is {shown_word}")
+
+
+    def get_letter_index(self, alpahabet_guessed: str) -> list:
+        index_alphabet = []
+        
+        for index, alphabet in enumerate(list(self.word)):
+            if alphabet == alpahabet_guessed:
+                index_alphabet.append(index)
+
+        return index_alphabet
+
+
+    def update_word(self,location: list, letter: str):
+        for index in location:
+            # print(self.word)
+            self.display[index] = letter
+
+
+    def is_already_guessed(self,letter: str) -> bool:
+        if letter not in self.guessed_letters:
+                self.guessed_letters.append(letter)
+                return False
+        return True
+
+
+    def check_guess(self, alpahabet_guessed: str):
+        if alpahabet_guessed in self.word:
+            if (not self.is_already_guessed(alpahabet_guessed)):
```

Comment:
we should try to keep one level nesting. two level nesting is still tolerable, however more than that is strongly discouraged. 

here we can do something like this to make it more readable

if self.is_alphabet_guessed_correctly_for_the_first_time(alpahabet_guessed):
     self.process_correctly_entered_alphabet_and_refresh_display(alpahabet_guessed)
     return

if self.is_alphabet_is_already_guessed():
     return

if self.is_alphabet_incorrectly_guessed():
     self.handle_incorrect_guess()